casper/src/main/resources/RevVault.rho

```diff
@@ -33,6 +33,17 @@ in {
       @MakeMint!(*mintCh) |
       for (mint <- mintCh) {
 
+        contract RevVault(@"deployerAuthKey", deployerId, ret) = {
+
+          new RevAddress(`rho:rev:address`), revAddrCh in {
+            RevAddress!("fromDeployerId", *deployerId, *revAddrCh)|
+            for (@deployerRevAddress <- revAddrCh) {
+
+              @AuthKey!("make", (*_revVault, deployerRevAddress), *ret)
```

Comment:
(this is kind of a note to self, but maybe you'll have some ideas?)

I'm a bit torn on using revAddress as part of the AuthKey.shape, but at the same time `findOrCreate` only receives a revAddress...

In my ideal world, AuthKey creation can be moved out of RevVault and be decoupled from it - the users would create an AuthKey of an agreed shape, based on a publicly available, bundled _revVault unf name (that encodes the contract we're creating an AuthKey for). 

That isn't safe though, if the shape is just `(bundle0{_revVault}, revAddressString)` (everybody has the rev address).

One direction for plugging that would be a `shape` of `(_deployerAuthKeyMarker, bundle0{_revVault}, revAddressString)`, where a key with a shape containing `_deployerAuthKeyMarker` would only be obtainable from a safe `AuthKey!deployersKey(deployerId, contractMarker, payload)`. This doesn't align seamlessly with unf-name based auth-keys though, as RevVault no longer knows the whole shape and would need to differentiate how to check the AuthKey (for deployerId vs unf-name based vaults).

Another approach would be somehow changing the internal account token from `revAddressString` to `bundle0{ deployerId }` / `bundle0 { unfName }`. The bundle is needed for RevVault to never know the `deployerId`, but at the same time it makes 3rd-party RevVault creation impossible... The corresponding revAddresses would be derived from deployerId / unfName, except not possible because a bundle wraps them....

It seems as though we need to leave `RevVault!deployerAuthKey` coupled with RevVault for now, and base our trust on it doing the right thing...

Reply:
I think this can be solved by making AuthKey create sth. like a pair of (priv, pub)-key - each of them would be a contract, pub would be used to verify the priv. We could have different methods for obtaining (parts of) the keypair (deployerId-based, unf-name-based) and only the public part of the keyparir would be needed upon creation.

Reply:
>I think this can be solved by making AuthKey create sth. like a pair of (priv, pub)-key - each of them would be a contract, pub would be used to verify the priv. We could have different methods for obtaining (parts of) the keypair (deployerId-based, unf-name-based) and only the public part of the keyparir would be needed upon creation.

Sounds like an interesting idea.
This would however force the user to use public keys instead of rev addresses for `findOrCreate`. Not sure if this is what we want.

Reply:
it's "pub key-like contract from AuthKey", it doesn't have to use anything else than the provided shape (which is arbitrary), and an AuthKey-private token to work. The private counterpart would require passing deployId + shape for construction, and would respond with `bundle0{(_authKey, shape)} when `challenge`d by the pub part :)

Reply:
There would be many variants of the `_authKey` token, to allow AuthKey-pairs derived from at least:
 - deployId
 - whatever unforgable name

Reply:
But how does the *create*-part of `findOrCreate` would work with this when the user that wants to create the rev vault is not the owner? The user would have to know about the "pub key-like contract from AuthKey" of the owner, correct? That might be a problem?

Reply:
The find-or-create method would create the public key. Creating the public authkey is safe ocap-leak-wise. It is the private part that is sensitive - we don't want to be passing the deployerId to too many places.

The above discussion's goal is to make the AuthKey!create*Private methods be the only places one has to pass such a high-grade auth ocap and obtain an attenuated, purpose specific ocap that can be checked using the corresponding public part, without the risk of leaking the high-grade ocap.